inc/features/blocks/personalization-variant/edit.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+const { InnerBlocks } = wp.blockEditor;
+const { compose } = wp.compose;
+const { withSelect, withDispatch } = wp.data;
+
+const Edit = ( {
+	hasChildBlocks,
+	isSelected,
+	selectParent,
+} ) => {
+	// Autofocus experience block variant parent on select.
+	if ( isSelected ) {
+		selectParent();
+	}
+
+	return (
+		<InnerBlocks
+			renderAppender={
+				hasChildBlocks
+					? undefined
+					: () => <InnerBlocks.ButtonBlockAppender />
+			}
+		/>
+	);
+};
+
+export default compose(
+	withSelect( ( select, ownProps ) => {
+		const { clientId } = ownProps;
+		const { getBlockOrder } = select( 'core/block-editor' );
+
+		return {
+			hasChildBlocks: () => getBlockOrder( clientId ).length > 0,
+		};
+	} ),
+	withDispatch( ( dispatch, ownProps, registry ) => {
+		const { clientId } = ownProps;
+		const { getBlockRootClientId } = registry.select( 'core/block-editor' );
+		const { selectBlock } = dispatch( 'core/block-editor' );
+
+		// Get parent block client ID.
+		const rootClientId = getBlockRootClientId( clientId );
+
+		return {
+			selectParent: () => selectBlock( rootClientId ),
```

Comment:
This should be named `onSelectParent` so it's obvious that it's a callback.

Reply:
It's not a callback, it's an action that selects the parent block when you call it.

Reply:
To be clear, I'm saying any callable (i.e. function) should be `on...` (as a form of Hungarian notation). This is a function that's passed in as a prop, so it should be prefixed with `on` even if it's not a user-facing action per se.

Reply:
Hmm, I see your point in terms of composing the component without the store but the only potential use-case for that is for testing and you'd still need to mock out the store which feels like a waste of time compared to running browser tests with e2e utilities.

I think when you're dealing with a redux store these components aren't meant to exist on their own in a reusable way and they don't need to. Some props will update and affect the other props.

Take the example of `setAttributes()` - it's not a function that is implicitly and reproducibly bound to some event by default. I would expect a `prop` called `onSetAttributes()` to be fired when `setAttributes()` is called.